main.py

```diff
@@ -783,6 +783,13 @@ def patch_record_in_place(fn, record, subdir):
                 depends[i] = "scipy >=1.7.1"
                 break
 
+    # Tensorflow numpy incompatibilites
+    if name.startswith("tensorflow-base"):
+        if VersionOrder(version) <= VersionOrder("2.6.0"):
+            replace_dep(depends, "numpy >=1.20", "numpy >=1.20,<2.0a0")
+        if VersionOrder(version) <= VersionOrder("2.5.0"):
+            replace_dep(depends, "numpy >=1.16.6,<2.0a0", "numpy >=1.16.6,<1.20.0a0")
```

Comment:
```suggestion
            replace_dep(depends, "numpy >=1.16.6,<2.0a0", "numpy >=1.16.6,<1.24.0a0")
```
The particular deprecation mentioned in #215 appears to have expired in numpy 1.24: https://numpy.org/devdocs/release/1.24.0-notes.html#expired-deprecations